### DISCUSSION TITLE
Comparison to Toppra, constant acceleration option?

### DISCUSSION THREAD

**Discussion by baichuan05** [NVlabs/curobo#276] [Software Q&A]
Hi!

I am trying to use curobo on ABB robot.
For the same start and goal, I have generated the figure below.

Left is the RRTConnect+Toppra and right is the curobo. At the first row, the curobo seems a bit slower than toppra in terms of execution time (after interploation). Is there a way to speed up the motion?

From what I saw, it seems that Toppra can use Constant acceleration so the execution will be faster. But curobo gives a more smooth acceleration change.
I tried to change jerk value, but it didn't seem working, to speed up.

![image](https://github.com/NVlabs/curobo/assets/20850928/4306e76e-7813-4d90-b6c4-10853e975e4c)

If you have any suggestions, that would be great! Thank you!

**Comment by peter-mitrano-bg** [comment]
Disclaimer I am not an author of curobo. Have you tried changing the interpolation_dt? or `finetune_dt_scale` ? My understanding is that if you reduce either value, and it still finds a solution, that solution will be faster (less motion time). curobo seems try to reduce motion time in the "traj opt fine tune" step, by scaling down the dt (by default, 90%) and the checking if it's still valid. If not, it increases dt. So, try reducing those two values:

```
# 1)

        self.motion_gen_config: MotionGenConfig = MotionGenConfig.load_from_robot_config(
            ...
            interpolation_dt=0.01,

# 2) 
        plan_cfg = MotionGenPlanConfig(
            ...
            finetune_dt_scale=0.7, # try adding this?
           
```
